concurrent_swiss_map.go

```diff
@@ -95,6 +95,16 @@ func (m *CsMap[K, V]) Has(key K) bool {
 	return shard.items.HasWithHash(key, hashShardPair.hash)
 }
 
+func (m *CsMap[K, V]) Clear() {
+	for i := 0; i < len(m.shards); i++ {
+		shard := m.shards[i]
+
+		shard.RLock()
```

Comment:
we shouldn't be able to update the map while cleaning, we need to use write lock here what do you think?

Reply:
😁 you are right, I modified this code based on range, this is my mistake.